### PR TITLE
feat: Added sharing functionality

### LINKED ISSUE #22: Sharing items
We need to implement sharing of items between users

### LINKED ISSUE #26: Fix CORS errors in ServiceWorker
https://github.com/HF6-PROJECT/web/pull/24/files

Implement this in ServiceWorker also
This might fix getting logged out.

### PR DESCRIPTION
**Pull request by frederikpyt**
## Description
- Fixed the CORS issue when using ngrok locally
- Added share action on both file & folder contextmenu
- Added "Shared with me" tab
- Made the active tab visually appear as active

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: src/components/item/SharingBrowser.vue
Comment (kristianbinau) [inline src/components/item/SharingBrowser.vue:1]: Shouldn't we remove the fetching from `Browser.vue` and just give the items to it through a prop.
Then we could have a `SharingBrowser.vue` & a `FileBrowser.vue` or something like that.

File: src/components/item/ShareItemModal.vue [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+	return colours[charCode % colours.length];
+}
+
+function getInitialsByName(name: string) {
+	const rgx = new RegExp(/(\p{L}{1})\p{L}+/, 'gu');
+
+	const initialsRegex = [...name.matchAll(rgx)] || [];
+
+	return ((initialsRegex.shift()?.[1] || '') + (initialsRegex.pop()?.[1] || '')).toUpperCase();
```
Comment (kristianbinau) [inline src/components/item/ShareItemModal.vue]: Make helper functions or something

File: src/components/item/ShareItemModal.vue [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+});
+
+defineExpose({
+	open,
+	close,
+});
+
+const usersWithAccess = ref<
+	{ id: number; name: string; initials: string; email: string; sharingId?: number }[]
+>([]);
```
Comment (kristianbinau) [inline src/components/item/ShareItemModal.vue]: Ugly TS

**Review by kristianbinau — CHANGES_REQUESTED** [review]
Haven't looked at design or tested yet. This is just looking at the code

File: src/components/item/SharingBrowser.vue
Comment (frederikpyt) [reply src/components/item/SharingBrowser.vue:1]: But the behavior is different. You shouldn't be able to open the "global" context menu in the SharingBrowser

**Review by kristianbinau — APPROVED** [review]
Awesome job, Comrade Pytlick🫡

**Pull request merged**